File: 백준/Silver/4423. Watching Watches/Watching Watches.py
import sys

def solve():
    lines = sys.stdin.readlines()

    for line in lines:
        if not line.strip():
            continue
        try:
            k, m = map(int, line.split())
        except ValueError:
            continue
        diff_per_day = abs(k - m)
        days_to_sync = 43200.0 / diff_per_day
        total_seconds_moved = days_to_sync * (84600.0 + (1800.0 - k))  # 86400 - k
        final_seconds = total_seconds_moved % 43200.0
        total_minutes = int((final_seconds + 30.0) // 60)
        total_minutes %= 720

        hours = total_minutes // 60
        mins = total_minutes % 60

        if hours == 0:
            hours = 12
        print(f"{k} {m} {hours:02d}:{mins:02d}")

if __name__ == "__main__":
    solve()

File: 백준/Silver/4423. Watching Watches/Watching Watches.cc
#include <bits/stdc++.h>
#define fastio() ios::sync_with_stdio(0); cin.tie(0); cout.tie(0)
using namespace std;

void solve() {
    int k, m;
    while (cin >> k >> m) {
        double diff_per_day = abs(k - m);
        double days_to_sync = 43200.0 / diff_per_day;
        double total_seconds_moved = days_to_sync * (86400.0 - (double)k);
        double final_seconds = fmod(total_seconds_moved, 43200.0);
        int total_minutes = (int)((final_seconds + 30.0) / 60.0);
        total_minutes %= 720;
        
        int hours = total_minutes / 60;
        int mins = total_minutes % 60;
        if (hours == 0) hours = 12;

        cout << k << " " << m << " " 
             << setfill('0') << setw(2) << hours << ":" 
             << setfill('0') << setw(2) << mins << "\n";
    }
}

int main() {
    fastio();
    solve();
    return 0;
}